README_cmake.md

```diff
@@ -1,106 +1,377 @@
-readily available from [Homebrew](https://brew.sh). Simply running
+Finally, on Ubuntu 20.04 LTS, you should install the following packages:
 
 ```
-$ brew update
-$ brew install cmake
+dev@ubuntu:~$ sudo apt install \
+                  clang-tools lld llvm-dev libclang-dev liblld-10-dev \
+                  libpng-dev libjpeg-dev libgl-dev \
+                  python3-dev python3-numpy python3-scipy python3-imageio python3-pybind11 \
+                  libopenblas-dev libeigen3-dev libatlas-base-dev \
+                  doxygen ninja-build
 ```
 
-should install the newest version of CMake.
+# Building Halide with CMake
 
-The [releases on Kitware's website](https://cmake.org/download/) are also viable
-options.
+## Basic build
 
-#### Linux
+These instructions assume that your working directory is the Halide repo root.
 
-If you're on **Ubuntu Linux 20.04** (focal), then simply running
-`sudo apt install cmake` is sufficient.
+### Windows
 
-For older versions of **Debian, Ubuntu, Mint, and derivatives**, Kitware
-provides an [APT repository](https://apt.kitware.com/) with up-to-date releases.
-Note that this is still useful for Ubuntu 20.04 because it will remain up to
-date.
+If you plan to use the Ninja generator, be sure to be in the developer command
+prompt corresponding to your intended environment. In particular, if you wish to
+compile a 32-bit version of Halide, you should use the 64-bit tools. You can
+configure the environment using the standard `vcvarsall.bat` script:
 
-For other Linux distributions, check with your distribution's package manager or
-use pip as detailed above. Snap packages are also available.
+```
+D:\> "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x64  (or x64_x84 for 32-bit)
+```
+
+Assuming that vcpkg is installed to `D:\vcpkg`, simply run:
 
-This applies to WSL (Windows Subsystem for Linux) installations, too.
+```
+> cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE=D:\vcpkg\scripts\buildsystems\vcpkg.cmake -S . -B build
+> cmake --build .\build
+```
+
+Otherwise, if you wish to create a Visual Studio based build system, you can
+configure with:
+
+```
+> cmake -G "Visual Studio 16 2019" -Thost=x64 -A x64 ^
+        -DCMAKE_TOOLCHAIN_FILE=D:\vcpkg\scripts\buildsystems\vcpkg.cmake ^
+        -S . -B build
+> cmake --build .\build --config Release -j %NUMBER_OF_PROCESSORS%
+```
 
-## Using Halide in JIT mode
+or, for 32-bit:
+
+```
+> cmake -G "Visual Studio 16 2019" -Thost=x64 -A Win32 ^
+        -DCMAKE_TOOLCHAIN_FILE=D:\vcpkg\scripts\buildsystems\vcpkg.cmake ^
+        -S . -B build
+> cmake --build .\build --config Release -j %NUMBER_OF_PROCESSORS%
+```
+
+### macOS and Linux
+
+The instructions here are straightforward. Assuming your environment is set up
+correctly, just run:
+
+```
+dev@host:~/Halide$ cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -S . -B build
+dev@host:~/Halide$ cmake --build ./build
```

Comment:
The section below on using Halide from your cmake build seems to require an install, but these instructions don't produce one, right? I typically need to set CMAKE_INSTALL_PREFIX or whatever it's called too.

Reply:
Good catch! I'll add those directions